components/match2/wikis/hearthstone/match_summary.lua

```diff
@@ -32,39 +38,170 @@ end
+			showLink = true,
+			overflow = 'ellipsis',
+		}
+	end
+
+	---@param opponentIndex any
+	---@return Html
+	local createScore = function(opponentIndex)
+		local isWinner = opponentIndex == game.winner or game.resultType == 'draw'
+		if game.resultType == 'default' then
+			return OpponentDisplay.BlockScore{
+				isWinner = isWinner,
+				scoreText = isWinner and 'W' or string.upper(game.walkover),
+			}
+		end
 
-	local team1Characters = Array.map((game.opponents[1] or {}).players or {}, Operator.property('character'))
-	local team2Characters = Array.map((game.opponents[2] or {}).players or {}, Operator.property('character'))
+		local score = game.resultType ~= 'np' and (game.scores or {})[opponentIndex] or nil
+		return OpponentDisplay.BlockScore{
+			isWinner = isWinner,
+			scoreText = score,
+		}
+	end
 
+	return HtmlWidgets.Div{
+		classes = {'brkts-popup-header-dev'},
+		css = {['justify-content'] = 'center', margin = 'auto'},
+		children = WidgetUtil.collect(
+			game.header and {
+				HtmlWidgets.Div{css = {margin = 'auto'}, children = {game.header}},
+				MatchSummaryWidgets.Break{},
+			} or nil,
+			HtmlWidgets.Div{
+				classes = {'brkts-popup-header-opponent', 'brkts-popup-header-opponent-left'},
+				children = {
+					createOpponent(1),
+					createScore(1):addClass('brkts-popup-header-opponent-score-left'),
+				},
+			},
+			HtmlWidgets.Div{
+				classes = {'brkts-popup-header-opponent', 'brkts-popup-header-opponent-right'},
+				children = {
+					createScore(2):addClass('brkts-popup-header-opponent-score-right'),
+					createOpponent(2),
+				},
+			}
+		)
+	}
+end
+
+---@param isTeamMatch boolean
+---@param game MatchGroupUtilGame
+---@param gameIndex number
+---@return Widget
+function CustomMatchSummary._createGame(isTeamMatch, game, gameIndex)
 	return MatchSummaryWidgets.Row{
 		classes = {'brkts-popup-body-game'},
-		css = {['font-size'] = '80%', padding = '4px', ['min-height'] = '24px'},
+		css = {padding = '4px', ['min-height'] = '24px'},
 		children = WidgetUtil.collect(
-			MatchSummaryWidgets.Characters{characters = team1Characters, flipped = false},
+			game.header and {
+				HtmlWidgets.Div{css = {margin = 'auto'}, children = {game.header}},
+				MatchSummaryWidgets.Break{},
+			} or nil,
+			CustomMatchSummary._displayOpponents(isTeamMatch, game.opponents[1].players, true),
 			MatchSummaryWidgets.GameWinLossIndicator{winner = game.winner, opponentIndex = 1},
+			MatchSummaryWidgets.GameCenter{css = {['font-size'] = '80%', ['flex'] = 1}, children = 'Game ' .. gameIndex},
 			MatchSummaryWidgets.GameWinLossIndicator{winner = game.winner, opponentIndex = 2},
-			MatchSummaryWidgets.Characters{characters = team2Characters, flipped = true}
+			CustomMatchSummary._displayOpponents(isTeamMatch, game.opponents[2].players)
 		)
 	}
 end
 
+---@param isTeamMatch boolean
+---@param players table[]
+---@param flip boolean?
+---@return Html?
+function CustomMatchSummary._displayOpponents(isTeamMatch, players, flip)
```

Comment:
could potentially use widgets here too

Reply:
Tried to use more widgets, if you have suggestion let me know